src/openms/include/OpenMS/DATASTRUCTURES/StringUtils.h

```diff
@@ -781,32 +781,27 @@ namespace OpenMS
     return this_s;
   }
 
-  static String& removeWhitespaces(String & this_s)
+  static String& removeWhitespaces(String& this_s)
   {
-    bool contains_ws = false;
-    for (std::string::const_iterator it = this_s.begin(); it != this_s.end(); ++it)
+    int shift(0); // windowing number of removed whitespaces
+    std::string::iterator it_end = this_s.end();
+    for (std::string::iterator it = this_s.begin(); it != it_end; ++it)
     {
-      char c = *it;
+      const char c = *it;
       if (c == ' ' || c == '\t' || c == '\n' || c == '\r')
       {
-        contains_ws = true;
-        break;
+        ++shift;
+        continue; // no need to shift this 
       }
-    }
-
-    if (contains_ws)
-    {
-      std::string tmp;
-      tmp.reserve(this_s.size());
-      for (std::string::const_iterator it = this_s.begin(); it != this_s.end(); ++it)
-      {
-        char c = *it;
-        if (c != ' ' && c != '\t' && c != '\n' && c != '\r')
-          tmp += c;
+      if (shift)
```

Comment:
why not use an iterator for the destination and use 
```
std::string::iterator dest = this_s.begin();
if (dest != it) *dest = *it;
```
and then increase `dest` and `it` together except if we hit a whitespace?

Reply:
I tried this, and performance-wise, there is no measureable difference.
So I guess its a matter of taste. Any strong feelings?

```
static String& removeWhitespaces2(String& this_s)
{
  std::string::iterator it = this_s.begin();
  std::string::iterator it_end = this_s.end();
  std::string::iterator dest = this_s.begin();
  while (it != it_end)
  {
    const char c = *it;
    if (c == ' ' || c == '\t' || c == '\n' || c == '\r')
    {
      ++it;
      continue;
    }
    if (dest != it) *dest = *it;
    ++dest;
    ++it;
  }

  // shorten result
  if (dest != it) this_s.resize(dest - this_s.begin());

  return this_s;
}
```